graylog2-web-interface/src/components/lookup-tables/CacheTableEntry.test.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { render, screen } from 'wrappedTestingLibrary';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import { asMock } from 'helpers/mocking';
+
+import { CACHE, mockedUseScopePermissions } from './fixtures';
+import CacheTableEntry from './CacheTableEntry';
+
+jest.mock('hooks/useScopePermissions', () => (mockedUseScopePermissions));
+
+const renderedCTE = (scope: string) => {
+  const auxCache = { ...CACHE };
+  auxCache._scope = scope;
+
+  return render(
+    <Router><CacheTableEntry cache={auxCache} /></Router>,
+    {
+      container: document.body.appendChild(document.createElement('table')),
+    },
+  );
+};
+
+describe('CacheTableEntry', () => {
+  it('should show Loading spinner while loading scope permissions', async () => {
+    asMock(mockedUseScopePermissions).mockReturnValueOnce({
+      loadingScopePermissions: true,
+      scopePermissions: null,
+    });
```

Comment:
This seems to work, but I can recommend a slightly cleaner approach:

It was confusing for me that the `fixtures` file contains a complete mock for the implementation of `useScopePermissions`.

Instead I would just mock `useScopePermissions` like this:
`jest.mock('hooks/useScopePermissions')`

and implement
```
beforeEach(() => {
  asMock(useScopePermissions).mockReturnValue(scopePermissions);
})
```
in this case `scopePermissions` is a fixture.

In the test you can then do
```
asMock(useScopePermissions).mockReturnValue({
  loadingScopePermissions: true,
  scopePermissions: null,
});
```
There are a few benefits
- `asMock(useScopePermissions).mockReturnValue` is easier to read
- by replacing `mockReturnValueOnce ` with `mockReturnValue` it behaves more like in the real world, because the state does not change just because another render happened. This way it is in general also easier to test something, because the state / return value is consistent.
- with the `beforeEach` part you can be sure that you reset the return value before every test.